test/utils.sh

```diff
@@ -1,13 +1,41 @@
 #!/bin/bash
-set -euo pipefail
+set -xeuo pipefail
 
 # lookup test-env.sh.yml for explanation.
 export KEY_VALUE_NOT_EXIST=" "
 
+wait_for_it() {
+  local timeout=$1
+  local spacer=2
+  shift
+
+  if ! [ $timeout = '-1' ]; then
+    local times_to_run=$((timeout / spacer))
+
+    #echo "Waiting for '$@' up to $timeout s"
+    for i in $(seq $times_to_run); do
+      eval $*  && return 0
+      sleep $spacer
+    done
+
+    # Last run evaluated. If this fails we return an error exit code to caller
+    eval $*
+  else
+    echo "Waiting for '$*' forever"
+
+    while ! eval $* > /dev/null; do
+      sleep $spacer
+    done
+  echo 'Success!'
+  fi
+}
+
 if [ $PLATFORM = "kubernetes" ]; then
-    cli=kubectl
+    cli="wait_for_it 600 kubectl"
+    cli_no_timeout=kubectl
```

Comment:
1. why do we need the separation? When will we need to wait and when won't we need to?
2. I think that we need to change `cli` to `cli_with_timeout`. It's not good to hide such logic.
3. Assuming we do need a separation, is there a reason not to write `wait_for_it 600 $cli` whenever we write it so it's clear to the reader that we wait?

Reply:
1. all the command will run with retry unless you handle / waiting for failure currently there is one command using the timeout (has_namespace and i added comment there)
2. sounds reasonable i will change it
3. User will have to know if use retry or not.. i want it will be the default act otherwise we will need to update each command and make sure new commands have the retry function as well. This is what was done in the past some of the commands have the retry and pass and the commands without the retry sometime failed.

Reply:
1. Sorry, i don't understand. If I understand correctly, `wait_for_it 600` waits *up to* 600 seconds. Thus, if a command doesn't need 600 seconds to run, it won't want 600 seconds. I still don't understand the case where waiting will cause an issue.
2. thanks.
3. I agree with your approach. If it's clear that there is `cli_with_timeout` & `cli_without_timeout` then it's ok to have the actual timeout logic inside. Please rename `cli_no_timeout` to `cli_without_timeout`

Reply:
1. The major issue in this job that you run command which suppose to work and from some reason it return no route problem. When i run the retry it will run the command again and will pass because the "no route" issue is not real one.
example 
`[2020-03-12T12:03:51.811Z] error: error when creating "STDIN": Post https://openshift-310.itci.conjur.net:8443/apis/apps.openshift.io/v1/namespaces/test-app-5-77cc69bd-c/deploymentconfigs: dial tcp 18.215.106.98:8443: connect: no route to host`
with retry the command will run again and pass

Reply:
which is good? do we want it to pass in the second time?

Reply:
👍